replicaset.js

```diff
@@ -0,0 +1,113 @@
+// Based on https://github.com/knex/knex/issues/2253#issuecomment-551610832
+
+const Knex = require('knex');
+const initKnex = require('./initKnex');
+
+exports.createRoundRobinSelectionStrategy = () => {
+    let round = {
+        true: 0,
+        false: 0,
+    };
+    return (pool, isWrite) => {
+        const selected = pool[round[isWrite]];
+        round[isWrite] = (round[isWrite] + 1) % (pool.length);
+        return selected;
+    };
+};
+
+exports.isWriteQuery = (query) => {
+    return ['insert', 'del', 'update'].includes(query.method);
+};
+
+exports.isWriteBuilder = (builder) => {
+    // Enable query context override: knex.select('*').queryContext({ replicaNode: 'write' | 'read' })
+    if (builder._queryContext && 'replicaNode' in builder._queryContext) {
+        return builder._queryContext === 'write';
```

Comment:
A typo I suppose :pencil2: 
```suggestion
        return builder._queryContext.replicaNode === 'write';
```